solutions/cf-609E.mdx

```diff
@@ -0,0 +1,179 @@
+---
+id: cf-609E
+source: CF
+title: Minimum spanning tree for each edge
+author: Chongtian Ma
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/22187)
+
+## Implementation 
+
+As mentioned in the official editorial, the answer for each edge is removing the heaviest edge in the MST in
+the path from vertex $u$ to $v$ and add the given weight. This will ensure that the MST is still connected.
+
+**Time Complexity:** $\mathcal{O}(M\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+const int MAXN = 2e5 + 1;
```

Comment:
MAXN is really 2e5, so either 0-index or use MAXN + 1 where needed instead

Reply:
well its the MAXN if one indexed, I can add a comment but i think using MAXN + 1 everywhere will be more confusing?

Reply:
or you could just, yknow, 0-index

Reply:
but ill be adding one to more things and things become uncool

Reply:
just sub 1 from the node indices?